### ISSUE TITLE
Suggestion implement the new duplication glitch

### ISSUE THREAD

**Issue by Paralusion** [THEREALWWEFAN231/Inertia#988]
**Minecraft Version(s)**
all
**What is the suggestion**
Implement the new duplication glitch, commonly used with the ".d" command, only need is a book and quil (you probably already know which dupe)
you should also put a disclaimer on the command, explaining that the dupe is experimental and that you can't use it without some knowledge about it
**Why would this feature be useful**
allows for duplicating items ingame

**Comment by constantinDev** [comment]
[Made a mod for that](https://github.com/AriliusClient/BookBot)
`@bb dupebook` ingame while holding a writable book

**Comment by GaiaB0t** [comment]
the point is for it to be in inertia itself

**Comment by Paralusion** [comment]
> [Made a mod for that](https://github.com/AriliusClient/BookBot)
> `@bb dupebook` ingame while holding a writable book

cool mod, but should be easy enough to be implemented into inertia as just another feature

**Comment by GaiaB0t** [comment]
should be really easy as book bot stuff is already in inertia

**Comment by constantinDev** [comment]
> the point is for it to be in inertia itself

The mod can be added to inertia easily, same version.

**Comment by DatSkyr3x** [comment]
what dupe are you talking about? do you mean the one where you have too much data in a chunk by having a bunch of books with ascii characters in them? that got patched and has  been public for a while

**Comment by Paralusion** [comment]
> what dupe are you talking about? do you mean the one where you have too much data in a chunk by having a bunch of books with ascii characters in them? that got patched and has been public for a while

no, the new dupe based on one book and quil being in your hand

**Comment by DatSkyr3x** [comment]
> no, the new dupe based on one book and quil being in your hand

didn't know about that, how does it work?

**Comment by DatSkyr3x** [comment]
> > what dupe are you talking about? do you mean the one where you have too much data in a chunk by having a bunch of books with ascii characters in them? that got patched and has been public for a while
> 
> no, the new dupe based on one book and quil being in your hand

i haven't seen that dupe before, can you explain how it works?

**Comment by AutistEren** [comment]
> > > what dupe are you talking about? do you mean the one where you have too much data in a chunk by having a bunch of books with ascii characters in them? that got patched and has been public for a while
> > 
> > 
> > no, the new dupe based on one book and quil being in your hand
> 
> i haven't seen that dupe before, can you explain how it works?

im no expert but
it sends a very big packet using the book and when the server reads it, it kicks you

**Comment by GaiaB0t** [comment]
> > > > what dupe are you talking about? do you mean the one where you have too much data in a chunk by having a bunch of books with ascii characters in them? that got patched and has been public for a while
> > > 
> > > 
> > > no, the new dupe based on one book and quil being in your hand
> > 
> > 
> > i haven't seen that dupe before, can you explain how it works?
> 
> im no expert but
> it sends a very big packet using the book and when the server reads it, it kicks you

it does what eren says, and it fails to save playerdata when you are kicked

**Comment by DatSkyr3x** [comment]
> > 
> 
> i haven't seen that dupe before, can you explain how it works?



> > > > what dupe are you talking about? do you mean the one where you have too much data in a chunk by having a bunch of books with ascii characters in them? that got patched and has been public for a while
> > > 
> > > 
> > > no, the new dupe based on one book and quil being in your hand
> > 
> > 
> > i haven't seen that dupe before, can you explain how it works?
> 
> im no expert but
> it sends a very big packet using the book and when the server reads it, it kicks you
> if you save ur inv by rejoining then put items in the chest and use the book, it thinks it already saved so when you rejoin you keep the items in ur inv and the chest
> idk if that was clear enough, go watch a video

i cant find any videos on it, only the old dupe where you overload a chunk

**Comment by GaiaB0t** [comment]
> > > 
> > 
> > 
> > i haven't seen that dupe before, can you explain how it works?
> 
> > > > > what dupe are you talking about? do you mean the one where you have too much data in a chunk by having a bunch of books with ascii characters in them? that got patched and has been public for a while
> > > > 
> > > > 
> > > > no, the new dupe based on one book and quil being in your hand
> > > 
> > > 
> > > i haven't seen that dupe before, can you explain how it works?
> > 
> > 
> > im no expert but
> > it sends a very big packet using the book and when the server reads it, it kicks you
> > if you save ur inv by rejoining then put items in the chest and use the book, it thinks it already saved so when you rejoin you keep the items in ur inv and the chest
> > idk if that was clear enough, go watch a video
> 
> i cant find any videos on it, only the old dupe where you overload a chunk
here
https://www.youtube.com/watch?v=ml3HkqNfUH4

**Comment by AutistEren** [comment]
https://youtu.be/zG-I6lKdUfU

**Comment by DatSkyr3x** [comment]
> https://youtu.be/zG-I6lKdUfU

thanks!

**Comment by constantinDev** [comment]
to everyone asking how it works:
1. book in hand
2. client sends a BIG bookupdate packet with a binary string
longer as the 16 bit int limit (32767), netty detects it, thinks its spam, kicks you
3. if the chunk is saved and you get removed, your inventory data isnt saved. Means if you do it correctly, you have the item in the chest and the item in your inventory from a previous inventory save

the dupe has since been patched ob 1.16.2+ spigot, bukkit, paper and vanilla.

**Comment by GaiaB0t** [comment]
> to everyone asking how it works:
> 
> 1. book in hand
> 2. client sends a BIG bookupdate packet with a binary string
>    longer as the 16 bit int limit (32767), netty detects it, thinks its spam, kicks you
> 3. if the chunk is saved and you get removed, your inventory data isnt saved. Means if you do it correctly, you have the item in the chest and the item in your inventory from a previous inventory save
> 
> the dupe has since been patched on 1.16.2+ spigot, bukkit, paper and vanilla.

that's not true, it works 1.16.3 paper and vanilla and 1.16.4 vanilla, spigot it never worked, idk about bukkit

**Comment by constantinDev** [comment]
> > to everyone asking how it works:
> 
> > 
> 
> > 1. book in hand
> 
> > 2. client sends a BIG bookupdate packet with a binary string
> 
> >    longer as the 16 bit int limit (32767), netty detects it, thinks its spam, kicks you
> 
> > 3. if the chunk is saved and you get removed, your inventory data isnt saved. Means if you do it correctly, you have the item in the chest and the item in your inventory from a previous inventory save
> 
> > 
> 
> > the dupe has since been patched on 1.16.2+ spigot, bukkit, paper and vanilla.
> 
> 
> 
> that's not true, it works 1.16.3 paper and vanilla and 1.16.4 vanilla, spigot it never worked, idk about bukkit

tried it in vanilla 1.15.2-1.15.4, didnt work. Neither with paper, bukkit or spigot.

**Comment by GaiaB0t** [comment]
what we are talking about, the .d one, works on 1.16.2 paper confirmed,people say it also works on older versions of paper 1.16.4, and does also work on paper 1.16.3

**Comment by GaiaB0t** [comment]
you might not be doing it right if it doesn't work for you

**Comment by constantinDev** [comment]
> you might not be doing it right if it doesn't work for you

i did it exactly as described in the vid, with the exact mod. didnt even kick me. so i made my own mod that sent more data and it kicked me, but didnt duplicate.
on vanilla, bukkit, spigot and paper, same results

**Comment by GaiaB0t** [comment]
you are holding the book, right?

**Comment by GaiaB0t** [comment]
when you type .d

**Comment by constantinDev** [comment]
ofc

**Comment by GaiaB0t** [comment]
then idk what the problem is it works for me fine

**Comment by GaiaB0t** [comment]
you have fabric api?

**Comment by constantinDev** [comment]
it doesnt require that but yes, i do

**Comment by constantinDev** [comment]
> then idk what the problem is it works for me fine

i think i may have an idea what is happening. Maybe they patched older jars too, so the bug doesnt work on older versions too. Or its just one specific plugin that triggers it idk

**Comment by DatSkyr3x** [comment]
> then idk what the problem is it works for me fine

i just tried it on a paper 1.16.4 aternos server and gave me the book too large error message and didnt dupe, then i tried running the latest paper 1.16.4 on my localhost and it was the same there, i think its already patched

**Comment by constantinDev** [comment]
> > then idk what the problem is it works for me fine
> 
> 
> 
> i just tried it on a paper 1.16.4 aternos server and gave me the book too large error message and didnt dupe, then i tried running the latest paper 1.16.4 on my localhost and it was the same there, i think its already patched
> 
> 

pog we got it patched before it was even big POGGGG

THANKS SKIDS OF YOUTUBE, LOVE YOU

**Comment by snoopchallenge** [comment]
I'm pretty sure this is an older dupe? Only worked because server software devs patched it incorrectly

**Comment by Godwhitelight** [comment]
wanna server that this dupe working on?

**Comment by 0x150** [comment]
> [Made a mod for that](https://github.com/AriliusClient/BookBot) `@bb dupebook` ingame while holding a writable book

bruh thats not your mod

**Comment by Autist69420** [comment]
> > [Made a mod for that](https://github.com/AriliusClient/BookBot) `@bb dupebook` ingame while holding a writable book
> 
> bruh thats not your mod
yikes, owner of it responded
